tools/participation-cli/main.go

```diff
@@ -67,52 +73,38 @@ func buildTransactionPayload(ctx context.Context, client *iotago.NodeHTTPAPIClie
 			return nil, err
 		}
 
-		if unspentOutput.Type() != iotago.OutputSigLockedSingleOutput {
+		if unspentOutput.Type() != iotago.OutputExtended {
 			continue
 		}
 
-		balance, err := unspentOutput.Deposit()
-		if err != nil {
-			return nil, err
-		}
-
-		inputsBalance += int(balance)
+		inputsBalance += unspentOutput.Deposit()
 		txBuilder.AddInput(&iotago.ToBeSignedUTXOInput{Address: inputAddress, Input: input})
-
-		if inputsBalance >= (int(outputAmount) + MINIMUM_DUST_BALANCE) {
-			// no need to collect further inputs
-			break
-		}
 	}
 
-	if inputsBalance < int(outputAmount) {
+	if inputsBalance < outputAmount {
 		return nil, fmt.Errorf("not enough balance on the inputs: %d, needed: %d", inputsBalance, outputAmount)
 	}
 
-	txBuilder.AddOutput(&iotago.SigLockedSingleOutput{Address: outputAddress, Amount: outputAmount})
-	inputsBalance -= int(outputAmount)
-
-	if inputsBalance != 0 && inputsBalance < MINIMUM_DUST_BALANCE {
-		return nil, fmt.Errorf("remainder does not fulfill the minimum balance requirement: %d, needed: %d", inputsBalance, MINIMUM_DUST_BALANCE)
-	}
+	txBuilder.AddOutput(&iotago.ExtendedOutput{Address: outputAddress, Amount: outputAmount})
+	inputsBalance -= outputAmount
 
 	if inputsBalance > 0 {
```

Comment:
Is there a way to determine the byte costs for an Extended output?
If yes, the remainder should be added to the output containing the participation if the remainder is less than the byte costs for the Extended output we create for the remainder.